recipe/multitenancy/multitenancyClaims.go

```diff
@@ -0,0 +1,84 @@
+package multitenancy
+
+import (
+	"errors"
+	"time"
+
+	"github.com/supertokens/supertokens-golang/recipe/multitenancy/multitenancyclaims"
+	"github.com/supertokens/supertokens-golang/recipe/session/claims"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+func NewMultitenancyClaims() (*claims.TypeSessionClaim, *claims.TypeSessionClaim, multitenancyclaims.TypeMultitenancyClaimValidators) {
+	fetchTenantId := func(userId string, userContext supertokens.UserContext) (interface{}, error) {
+		instance, err := GetRecipeInstanceOrThrowError()
+		if err != nil {
+			return nil, err
+		}
+		tenantIdRes, err := instance.GetTenantIdForUserID(userId, userContext)
+		if err != nil {
+			return false, err
+		}
+		if tenantIdRes.OK != nil {
+			return tenantIdRes.OK.TenantId, nil
+		} else {
+			return "", errors.New("UNKNOWN_USER_ID")
+		}
+	}
+
+	fetchDomains := func(userId string, userContext supertokens.UserContext) (interface{}, error) {
+		instance, err := GetRecipeInstanceOrThrowError()
+		if err != nil {
+			return nil, err
+		}
+		tenantIdRes, err := instance.GetTenantIdForUserID(userId, userContext)
+		if err != nil {
+			return false, err
+		}
+		if tenantIdRes.OK != nil {
+			tenantId := tenantIdRes.OK.TenantId
+			domains, err := instance.GetDomainsForTenantId(tenantId, userContext)
+			if err != nil {
+				return nil, err
+			}
+			return domains, nil
+		} else {
+			return nil, errors.New("UNKNOWN_USER_ID")
+		}
+	}
+
+	var defaultMaxAge int64 = 300
+	mtDomainClaim, arrayClaimValidators := claims.PrimitiveArrayClaim("st-tenant-domains", fetchDomains, &defaultMaxAge)
+	mtTenantIdClaim, _ := claims.PrimitiveClaim("st-tenantid", fetchTenantId, &defaultMaxAge)
+
+	getLastRefetchTime := func(payload map[string]interface{}, userContext supertokens.UserContext) *int64 {
+		if value, ok := payload[mtDomainClaim.Key].(map[string]interface{}); ok {
+			switch t := value["t"].(type) {
+			case int64:
+				return &t
+			case float64:
+				it := int64(t)
+				return &it
+			}
+		}
+		return nil
+	}
```

Comment:
isn't this something that is already impelemnted in the base claim? Why does this need impl again?

Reply:
Have removed this, we can stick to the default impl.